common/txmgr/confirmer.go

```diff
@@ -294,178 +260,155 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Pro
-//
-// 0. We are connected to multiple primary nodes via load balancer
-// 1. We send a transaction, it is confirmed and, we get a receipt
-// 2. A new head comes in from RPC node 1 indicating that this transaction was re-org'd, so we put it back into unconfirmed state
-// 3. We re-send that transaction to a RPC node 2 **which hasn't caught up to this re-org yet**
-// 4. RPC node 2 still has an old view of the chain, so it returns us "sequence too low" indicating "no problem this transaction is already mined"
-// 5. Now the transaction is marked "confirmed_missing_receipt" but the latest chain does not actually include it
-// 6. Now we are reliant on the Resender to propagate it, and this transaction will not be gas bumped, so in the event of gas spikes it could languish or even be evicted from the mempool and hold up the queue
-// 7. Even if/when RPC node 2 catches up, the transaction is still stuck in state "confirmed_missing_receipt"
-//
-// This scenario might sound unlikely but has been observed to happen multiple times in the wild on Polygon.
-func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CheckConfirmedMissingReceipt(ctx context.Context) (err error) {
-	attempts, err := ec.txStore.FindTxAttemptsConfirmedMissingReceipt(ctx, ec.chainID)
-	if err != nil {
-		return err
-	}
-	if len(attempts) == 0 {
+func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) ProcessReorgTxs(ctx context.Context, reorgTxs []*txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], head types.Head[BLOCK_HASH]) error {
+	if len(reorgTxs) == 0 {
 		return nil
 	}
-	ec.lggr.Infow(fmt.Sprintf("Found %d transactions confirmed_missing_receipt. The RPC node did not give us a receipt for these transactions even though it should have been mined. This could be due to using the wallet with an external account, or if the primary node is not synced or not propagating transactions properly", len(attempts)), "attempts", attempts)
-	txCodes, txErrs, broadcastTime, txIDs, err := ec.client.BatchSendTransactions(ctx, attempts, int(ec.chainConfig.RPCDefaultBatchSize()), ec.lggr)
-	// update broadcast times before checking additional errors
-	if len(txIDs) > 0 {
-		if updateErr := ec.txStore.UpdateBroadcastAts(ctx, broadcastTime, txIDs); updateErr != nil {
-			err = fmt.Errorf("%w: failed to update broadcast time: %w", err, updateErr)
+	var etxIDs, attemptIDs []int64
+	for _, etx := range reorgTxs {
+		if len(etx.TxAttempts) == 0 {
+			return fmt.Errorf("invariant violation: expected tx %v to have at least one attempt", etx.ID)
 		}
-	}
-	if err != nil {
-		ec.lggr.Debugw("Batch sending transactions failed", "err", err)
-	}
-	var txIDsToUnconfirm []int64
-	for idx, txErr := range txErrs {
-		// Add to Unconfirm array, all tx where error wasn't TransactionAlreadyKnown.
-		if txErr != nil {
-			if txCodes[idx] == client.TransactionAlreadyKnown {
-				continue
-			}
+
+		// Rebroadcast the one with the highest gas price
```

Comment:
just curious, is TxAttempts ordered

Reply:
We do in [LoadTxesAttempts](https://github.com/smartcontractkit/chainlink/blob/a3f97b85764f5862437247ea4d03219279588586/core/chains/evm/txmgr/evm_tx_store.go#L644). Whenever we load attempts, we order them latest to oldest using the gas price/gas tip cap in descending order